packages/next/server/render.tsx

```diff
@@ -1539,18 +1551,11 @@ export async function renderToHTML(
           renderStream,
           suffix,
           dataStream: serverComponentsInlinedTransformStream?.readable,
-          generateStaticHTML:
-            generateStaticHTML || !process.env.__NEXT_REACT_ROOT,
+          generateStaticHTML,
```

Comment:
Is it true that this will always be enabled for React 17 and below?

Reply:
There's a condition branch checking `__NEXT_REACT_ROOT` above, if it's using react 17, will go to another `if` branch. Here react root is always `true` then `|| !process.env.__NEXT_REACT_ROOT` is always `false`.

I also added some comments for `__NEXT_REACT_ROOT ` condition in the if-else branch